src/WebJobs.Extensions.DurableTask/SystemEnvironment.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Provides utilities for inspecting the process enviroment and platform, currently mostly
+    /// for checking if we are running on a Linux enviroment or not.
+    /// </summary>
+    /// Mostly duplicated functionality from:
+    /// https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Environment/EnvironmentExtensions.cs
+    /// We should consider abstracting away this functionality from both repos, to avoid code duplication.
+    internal class SystemEnvironment
```

Comment:
This seems like a lot of abstraction just to read some environment variables. I'm curious to understand why the Functions team decided to do it this way.

Reply:
I'll reach out to see what they say

Reply:
Although, from my understanding, it appears that their file is doing a lot more than just enviroment variable reading for them. I also think this design somehow facilitates testing for them, if I recall correctly

**The Functions version of this file:** https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Environment/EnvironmentExtensions.cs

Reply:
I know the functions team has done a lot of abstractions around the environment variables and file system to put them behind interfaces that they code against so they can easily mock them for unit tests. Without also introducing their infrastructure for things like IEnvironment, it is unlikely that we will see the same testing benefits here.

Reply:
That sounds about right to me.
That being said, and looking at the code in **our** `SystemEnviroment.cs`, I'm not immediately convinced it's abusing abstractions that much. It's certainly verbose, but the alternative would be to just inline a bunch of these calls into two functions instead of four small ones? Not sure if we're gaining that much by it, and I could imagine this utility file growing in the future.

I don't feel strongly though!

Reply:
I think this would be cleaner as a set of helper static methods for `INameResolver`. That way we can make this super unit testable by mocking `INameResolver`.